cucumber/authenticators_jwt/features/authn_jwt.feature

```diff
@@ -38,25 +38,56 @@ Feature: JWT Authenticator - JWKs Basic sanity
       member: !host myapp
     """
     And I am the super-user
-    And I successfully set authn-jwt jwks-uri variable with value of "myJWKs.json" endpoint
+    And I successfully set authn-jwt jwks-uri variable with value of "first.json" endpoint
     And I successfully set authn-jwt token-app-property variable to value "user"
 
   Scenario: A valid JWT token with identity in the token
     Given I have a "variable" resource called "test-variable"
     And I permit host "myapp" to "execute" it
     And I add the secret value "test-secret" to the resource "cucumber:variable:test-variable"
-    And I issue a JWT token:
+    And I issue a JWT token using key "first.json"::
     """
     {
       "user":"host/myapp",
       "project-id": "myproject"
     }
     """
     And I save my place in the audit log file
-    When I authenticate via authn-jwt with the JWT token
+    When I authenticate via authn-jwt with the JWT token issued by "first.json" key
     Then host "myapp" has been authorized by Conjur
     And I successfully GET "/secrets/cucumber/variable/test-variable" with authorized user
     And The following appears in the audit log after my savepoint:
     """
     cucumber:host:myapp successfully authenticated with authenticator authn-jwt service cucumber:webservice:conjur/authn-jwt/raw
     """
+
+  Scenario: Update jwks-uri dynamically
+    Given I successfully set authn-jwt jwks-uri variable with value of "NotFound.json" endpoint
+    And I issue a JWT token using key "first.json"::
+    """
+    {
+      "user":"host/myapp",
+      "project-id": "myproject"
+    }
+    """
+    And I save my place in the audit log file
+    When I authenticate via authn-jwt with the JWT token issued by "first.json" key
+    Then it is unauthorized
+    And The following appears in the audit log after my savepoint:
+    """
+    CONJ00185E
+    """
+
+    Given I initialize JWKs endpoint with file "second.json"
+    And I successfully set authn-jwt jwks-uri variable with value of "second.json" endpoint
+    And I save my place in the audit log file
+    When I authenticate via authn-jwt with the JWT token issued by "first.json" key
+    Then it is unauthorized
+    And The following appears in the audit log after my savepoint:
+    """
+    CONJ00035E
+    """
+
+    Given I successfully set authn-jwt jwks-uri variable with value of "first.json" endpoint
+    When I authenticate via authn-jwt with the JWT token issued by "first.json" key
+    Then host "myapp" has been authorized by Conjur
```

Comment:
Some comments here:
1. Looks like we have 2 different tests here and i think its better to separate between them. **Why?** If the first scenario success and the second one fail it doesnt mean the dynamic is not working.
2. Given is relevant only before we start the test, if you want to verify more issues it will be with `When` after the `Then`.  **Why?** If you have failure on `Given` it means something was failed before we start the test , with pre configure. If something was failed during the `When` it means during the test it self it failed. If something was failed during `Then` it means you didnt get what you want
3. I always prefer to verify the full audit log . **Why?** If we change the log i want the test will fail if we not fix it, and its more readable to the user while reading the test